icu4j/main/classes/core/src/com/ibm/icu/impl/units/MeasureUnitImpl.java

```diff
@@ -754,8 +754,9 @@ public MeasureUnitImplComparator(ConversionRates conversionRates) {
 
         @Override
         public int compare(MeasureUnitImpl o1, MeasureUnitImpl o2) {
-            UnitConverter fromO1toO2 = new UnitConverter(o1, o2, conversionRates);
-            return fromO1toO2.convert(BigDecimal.valueOf(1)).compareTo(BigDecimal.valueOf(1));
+            BigDecimal factor1 = this.conversionRates.getFactorToBase(o1).getConversionRate();
+            BigDecimal factor2 = this.conversionRates.getFactorToBase(o2).getConversionRate();
+            return factor1.compareTo(factor2);
```

Comment:
Found the tests by changing `return -factor1.compareTo(factor2);`
causes failures in testComplexUnitConverterSorting, testComplexUnitsConverter, and testUnitPreferencesWithCLDRTests in UnitsTest.
In NumberFormatterApiTest, there are also nice failures (in unitMeasure and unitUsage).

I don't find any failures in com.ibm.icu.dev.test.format.MeasureUnitTest, which does mean MeasureUnitImpl is not tested thoroughly in MeasureUnitTest.

Reply:
Probably makes sense: MeasureUnitImpl isn't generally tested, and the comparator is only used by complex conversion. In C++, measfmttest.cpp has only one test touching on MeasureUnitImpl, and it's the one I added to check on the leak.

Reply:
I'm marking this as resolved, I've repeated myself a bit in a comment below.